lib/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -0,0 +1,41 @@
+module Foreman
+  module Renderer
+    module Scope
+      module Macros
+        module SnippetRendering
+          def snippet_if_exists(name, options = {})
+            snippet(name, { silent: true }, variables: options[:variables])
+          end
+
+          # provide embedded snippets support as simple erb templates
+          def snippets(file, options = {})
+            if ::Template.where(:name => file, :snippet => true).empty?
+              render :partial => "unattended/snippets/#{file}"
+            else
+              snippet(file.gsub(/^_/, ""), options)
+            end
+          end
+
+          def snippet(name, options = {}, variables: {})
+            if (template = ::Template.where(:name => name, :snippet => true).first)
+              begin
+                snippet_variables = variables.merge(options[:variables] || {})
+                return Foreman::Renderer.render_template(template: template,
+                                                         subjects: { host: @host },
+                                                         variables: snippet_variables)
```

Comment:
:clap: this replaces the need for `@kept_variables` hack and is first fruit of the effort